conda-store-server/conda_store_server/orm.py

```diff
@@ -254,7 +255,12 @@ def build_path(self, conda_store):
         # https://github.com/conda-incubator/conda-store/issues/649
         if len(str(res)) > 255:
             raise BuildPathError("build_path too long: must be <= 255 characters")
-        return res
+        # Note: cannot use the '/' operator to prepend the extended-length
```

Comment:
Hm, this is surprising. Why? Do you have a reference? I would have assumed that this is ok:

```python
pathlib.Path(r"\\?\", res)
```

Reply:
Hmm, I cannot reproduce this anymore for some reason! Not sure what I've tried. E.g., this works fine:

```py
>>> str("\\\\?\\" / pathlib.Path("foo/bar"))
'\\\\?\\foo\\bar'
```

Reply:
Ah, I see what the problem is. In my comment above, the `Path` part is a relative path. But in the above code, `res` is absolute and it just ignores the prefix (I've printed the value to check):

```py
>>> str("\\\\?\\" / pathlib.Path("foo"))
'\\\\?\\foo'  # ok, but Path is relative
>>> str("\\\\?\\" / pathlib.Path("C:\\foo")) 
'C:\\foo'  # ignored
>>> str("\\\\?\\" / pathlib.WindowsPath("C:\\foo")) 
'C:\\foo'  # ignored (it's WindowsPath on Windows)
```

Same with your variant:

```py
>>> str(pathlib.Path("\\\\?\\", pathlib.Path("foo")))
'\\\\?\\foo'  # ok, but Path is relative
>>> str(pathlib.Path("\\\\?\\", pathlib.Path("C:\\foo")))
'C:\\foo'  # ignored
>>> str(pathlib.Path("\\\\?\\", pathlib.WindowsPath("C:\\foo")))
'C:\\foo'  # ignored
```

So I'll keep my current version with string interpolation.

Reply:
Ah, I see. Actually, it [looks like](https://stackoverflow.com/a/13426839) `\\?\` can _only_ be used with absolute paths. So maybe we want to add a .resolve() call below.

Reply:
nvm it's abspath'd above. Marking this as resolved. Thanks!